Automattic/wp-calypso
# Plans 2023: Add site-purchases (queries and selector hooks) to data-stores & use in pricing meta

Related to https://github.com/Automattic/wp-calypso/issues/79005

## Proposed Changes

Part of migrating the plan data layer to data-stores for querying outside of Calypso. As of https://github.com/Automattic/wp-calypso/issues/82761 we are querying the site purchases to determine the price a plan was purchased on (and use that for the Spotlight grid plan). This PR aims to extract the relevant pieces to query site purchases from the data-stores package, in effect removing the dependency and contributing towards extracting the rest of purchases state logic from Calypso into this more reusable form.

* Introduces `purchases` to data-stores package with queries and selector hooks for retrieving site purchases
* Main types are migrated to the package and re-imported into Calypso state controllers
* Only implements `site purchases` queries & selector hooks (there are also user purchases)

## Testing Instructions



- Confirm UI based on site purchases renders correctly where they are used
- Set currency to USD and add a non-standard plan to a site in SA e.g. Personal 2021
- Confirm the correct price is shown in `/plan/[ site ]` in the Spotlight card (should be the "purchase price", not the actual plan's current price)

## Pre-merge Checklist
